src/mlpack/bindings/python/mlpack/matrix_utils.py

```diff
@@ -55,7 +55,10 @@ def to_matrix(x, dtype=np.double, copy=False):
   else:
     if isinstance(x, pd.core.series.Series):
       x = pd.DataFrame(x)
```

Comment:
Do you think there is any way we could avoid copying the `Series` object?  Also, another thing to notice is that when we do `x = pd.DataFrame(x)`, we are already copying the DataFrame, so if `copy=True` we don't need to copy a second time.

Reply:
hi @rcurtin can we use `y = x.to_numpy()`

Reply:
Have you tried it to see if it causes a copy?

Reply:
Hi @rcurtin it doesn't make copy. if we run this 
`import pandas as pd
import numpy as np
d = pd.read_csv("some_data.csv")
x = d.to_numpy()
x.flags`

its gives 
`  C_CONTIGUOUS : False                                                                                                                                          
  F_CONTIGUOUS : True                                                                                                                                           
  OWNDATA : False                                                                                                                                               
  WRITEABLE : True                                                                                                                                              
  ALIGNED : True                                                                                                                                                
  WRITEBACKIFCOPY : False                                                                                                                                       
  UPDATEIFCOPY : False`


Here   `F_CONTIGUOUS`  is True we can resolve this by ` np.ma.array(x,order= 'C')`

Reply:
I see---but why are you using the masked array support, and not just `np.array(x, order='C')`?  It seems to me like the mask support isn't needed.

In any case, whatever we do, we should add a test case ensuring that passing a Pandas Dataframe (or Series) to an mlpack binding works.  You could imitate the tests in `src/mlpack/bindings/python/tests/test_python_binding.py` for regular numpy matrices.  I'd prefer to ensure that this code works before we merge it, because it's really easy to introduce segfaults, etc., unless we have thoroughly tested it.

Reply:
As you told me `OWNDATA = True` implies that the matrix was copied. when i was using `np.array(x,order= 'C').flags`  it gives me 
  `
  C_CONTIGUOUS : True                                                                                                                                          
  F_CONTIGUOUS : False                                                                                                                                           
  OWNDATA : True                                                                                                                                                
  WRITEABLE : True                                                                                                                                              
  ALIGNED : True                                                                                                                                                
  WRITEBACKIFCOPY : False                                                                                                                                       
  UPDATEIFCOPY : False
`
hence it implies that the matrix was copied.
 But When I was using `np.ma.array(x,order ='C').flags`
its gives me 
  `
  C_CONTIGUOUS : True                                                                                                                                          
  F_CONTIGUOUS : False                                                                                                                                           
  OWNDATA : False                                                                                                                                                
  WRITEABLE : True                                                                                                                                              
  ALIGNED : True                                                                                                                                                
  WRITEBACKIFCOPY : False                                                                                                                                       
  UPDATEIFCOPY : False
`
Hence it implies that the matrix was not copied.

Reply:
And I m working on those test cases.

Reply:
Let's not use the masked array support unless we can understand why that does not cause a copy.  It would be better if we could use `np.array()` and avoid a copy.  You may have to dig into the internals of numpy somewhat to learn more here.

Reply:
Hi we can use `y = np.ndarray(x.shape,buffer=x.flatten(),order = 'C')`  where x is  `x = d.to_numpy()`  when i use `y.flags` it gives me
 ```

  C_CONTIGUOUS : True
  F_CONTIGUOUS : False
  OWNDATA : False
  WRITEABLE : True
  ALIGNED : True
  WRITEBACKIFCOPY : False
  UPDATEIFCOPY : False
```

and its type is `numpy.ndarray`